### DISCUSSION TITLE
Py3.10 Support

### DISCUSSION THREAD

**Discussion by pm5k** [python-poetry/poetry#4608] [Q&A]
Hello, not sure if this was asked before, but I would like to know what is the status of support for python 3.10 as it has now been officially released? I tried to set up afresh by uninstalling poetry completely, then installing 3.10 with pyenv, setting it as the global version, reloading the shell (to ensure 3.10 is picked up globally) and then installing poetry using the recommended method: `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python -`

After this, simply calling `poetry` results in the following:

```
Traceback (most recent call last):
  File "/home/tom/.poetry/bin/poetry", line 17, in <module>
    from poetry.console import main
  File "/home/tom/.poetry/lib/poetry/console/__init__.py", line 1, in <module>
    from .application import Application
  File "/home/tom/.poetry/lib/poetry/console/application.py", line 3, in <module>
    from cleo import Application as BaseApplication
ModuleNotFoundError: No module named 'cleo'
```

There was an Issue - #3187 which explains that one could do:
```
pushd ~/.poetry/lib/poetry/_vendor/
ln -s py3.9 py3.10
```

But this seems hacky in a way. Is there some sort of timeline on when proper 3.10 support will be available as it seems that simply installing poetry this way ends up making it unusable unless the above symlink is created to "fix" the issue.

Thanks in advance.

**Comment by uSpike** [comment]
Hm, I tried this with 3.10 on my workstation and it works fine:
```shell
$ pyenv install 3.10.0
$ cd $(mktemp -d)
$ pyenv local 3.10.0
$ mkdir poetry_home
$ curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | POETRY_HOME=$(pwd)/poetry_home python3.10 
$ poetry_home/bin/poetry --version
Poetry version 1.1.11 
$ poetry_home/venv/bin/python --version
Python 3.10.0
```
Your error seems related to using `get-poetry.py` which vendors packages with specific versions of python (not including 3.10).  Are you sure you used `install-poetry.py`?
